src/utils/utils.ts

```diff
@@ -152,6 +150,35 @@ export async function getGitBranchName() {
   return gitApi.repositories[0]?.state.HEAD?.name;//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
 }
 
+function extractRepoFullName(remoteURL: string): string | undefined {
+  const match = remoteURL.match(/[:/]([^/]+)\/([^/]+?)(?:\.git)?$/);
+  return match ? `${match[1]}/${match[2]}` : undefined;
+}
+
+export async function getActiveRepository(): Promise<Repository | undefined> {
+  const gitAPI = await getGitAPIRepository();
```

Comment:
add try-catch here

Reply:
does not need, this function does not thrown error. it returns undefined if there is a problem